samples/Shared/SharedModels/Person.cs

```diff
@@ -15,16 +16,21 @@ public class PersonValidator : AbstractValidator<Person>
     {
         public PersonValidator()
         {
-            RuleFor(p => p.Name).NotEmpty().WithMessage("You must enter a name");
-            RuleFor(p => p.Name).MaximumLength(50).WithMessage("Name cannot be longer than 50 characters");
+            RuleSet("Names", () => {
+                RuleFor(p => p.FirstName).NotEmpty().WithMessage("You must enter your first name");
+                RuleFor(p => p.FirstName).MaximumLength(50).WithMessage("First name cannot be longer than 50 characters");
+                RuleFor(p => p.LastName).NotEmpty().WithMessage("You must enter your last name");
+                RuleFor(p => p.LastName).MaximumLength(50).WithMessage("Last name cannot be longer than 50 characters");
+            });
+           
             RuleFor(p => p.Age).NotNull().GreaterThanOrEqualTo(0).WithMessage("Age must be greater than 0");
             RuleFor(p => p.Age).LessThan(150).WithMessage("Age cannot be greater than 150");
             RuleFor(p => p.EmailAddress).NotEmpty().WithMessage("You must enter a email address");
             RuleFor(p => p.EmailAddress).EmailAddress().WithMessage("You must provide a valid email address");
 
-            RuleFor(x => x.Name).MustAsync(async (name, cancellationToken) => await IsUniqueAsync(name))
-                                .WithMessage("Name must be unique")
-                                .When(person => !string.IsNullOrEmpty(person.Name));
+            RuleFor(x => x.EmailAddress).MustAsync(async (name, cancellationToken) => await IsUniqueAsync(name))
```

Comment:
This is just a nit pick, but in the `IsUniqueAsync` method can it check against an email address instead of just a name so it makes sense with the new changes. Once that is done I'll merge this.